draft-irtf-cfrg-vdaf.md

```diff
@@ -1190,6 +1190,213 @@ how the nonces are chosen, but security requires that the nonces be unique. See
+
+* In general, each request includes the Leader's prep share for the previous
+  round and/or the prep message for the current round; correspondingly, each
+  response consists of the prep message for the current round and the Helper's
+  prep share for the next round.
+
+The Aggregators proceed in this ping-ponging fashion until a step of the
+computation fails, indicating the report is invalid and should be rejected, or
+the last round is reached and each Aggregator computes an output share. All
+told there there are `ceil((Vdaf.ROUNDS+1)/2)` requests sent.
+
+Each Aggregator handles these messages according to the following transition
+rule, so long as they are in the `Continued` state:
+
+~~~ transition
+def ping_pong_next(Vdaf,
+                   is_leader: bool,
+                   agg_param: Vdaf.AggParam,
+                   state: State,
+                   inbound: Optional[bytes],
+                   ) -> (State, Optional[PingPongMessage]):
+    round, host_prep_state = state.round, state.prep_state
+    if inbound != None:
+        if round == 0:
+            if is_leader:
+                return (Rejected(), None)
+            else:
+                prep_msg, peer_prep_share = None, inbound
+        elif round == Vdaf.ROUNDS:
+            prep_msg, peer_prep_share = inbound, None
+        else:
+            try:
+                prep_msg, peer_prep_share = decode_continue(inbound)
+            except:
+                return (Rejected(), None)
+    elif is_leader:
+        prep_msg, peer_prep_share = None, None
+    else:
+        return (Rejected(), None)
+
+    try:
+        # If `prep_msg == None` then this is the start of the
+        # first round. Otherwise, `prep_msg` is the prep message
+        # from the end of the previous round.
+        out = Vdaf.prep_next(host_prep_state, prep_msg)
+        round += 1
+        if type(out) == Vdaf.OutShare:
+            return (Finished(out), None)
+        (host_prep_state, host_prep_share) = out
+
+        if peer_prep_share != None:
+            prep_shares = [peer_prep_share, host_prep_share]
+            if is_leader:
+                prep_shares.reverse()
+            prep_msg = Vdaf.prep_shares_to_prep(
+                agg_param,
+                prep_shares,
+            )
+            out = Vdaf.prep_next(host_prep_state, prep_msg)
+            round += 1
+            if type(out) == Vdaf.OutShare:
+                return (Finished(out), prep_msg)
+            (host_prep_state, host_prep_share) = out
+
+        if is_leader and inbound == None:
+            outbound = host_prep_share
+        else:
+            outbound = encode_continue(prep_msg, host_prep_share)
+        return (Continued(round, host_prep_state), outbound)
+    except:
+        return (Rejected(), None)
+~~~
+
+Methods `encode_continue()` and `decode_contine()` are used to coalesce a prep
+message and the next prep share:
+
+~~~
+def encode_method(prep_msg: bytes, prep_share: bytes):
```

Comment:
nit: `encode_method` should be `encode_continue` I think (and similarly below)

Reply:
These methods have been deleted.